tests/functional/helpers/local-monorepo.ts

```diff
@@ -50,6 +50,12 @@ export default class LocalMonorepo<
     this.#workspaces = workspaces;
   }
 
+  async create() {
+    await super.create();
+    await this.runCommand('yarn', ['set', 'version', 'stable']);
+    await this.runCommand('yarn');
+  }
```

Comment:
`yarn workspaces list` requires Yarn 3. I'm not sure if there's a better way to handle this.

Reply:
True, this change would make this tool usable only on Yarn v3 projects. I am okay with this, but it is inconsistent with recent decisions, so it could create possible confusion if we want to expand the scope of this tool.

A bit of backstory: When we converted the `core` repo to a monorepo we decided to follow after the Snaps monorepo and use Yarn v3. After that, we effectively decided that Yarn v3 was the future and updated `action-publish-release` and `action-npm-publish` to only support v3. Libraries that were using v1 could continue use older versions of these actions and eventually once they supported v3, everything would be fine.

However, it sounds like, based on some recent changes, this plan may not be as practical as we'd hoped, because I see that we've effectively restored Yarn v1 support for polyrepos to `action-publish-release` (https://github.com/MetaMask/action-publish-release/pull/83) and `action-npm-publish` (https://github.com/MetaMask/action-npm-publish/pull/51/files).

So that leaves the question of what to do here. On one hand, `create-release-branch` is only used for `core` at the moment, which does support Yarn v3, and if we want to use this for Snaps too, then that would work fine since it uses v3 as well. So, one route is to change this tool to explicitly support v3 only.

However, if anyone wanted to extend this tool to support polyrepo packages, we would be back in the same boat we're in now, and we'd have to come up with a Yarn v1-compatible solution. Technically we already have one: [`getWorkspaceLocations` in `action-utils`](https://github.com/MetaMask/action-utils/blob/b33c0b6b5c6cb3d544f81231f6187a2925841ff5/src/package-utils.ts#L280). So, we could use this implementation instead of or even as an alternative to `yarn workspaces list`. 

What is your opinion?

Reply:
> A bit of backstory: When we converted the core repo to a monorepo we decided to follow after the Snaps monorepo and use Yarn v3. After that, we effectively decided that Yarn v3 was the future and updated action-publish-release and action-npm-publish to only support v3. Libraries that were using v1 could continue use older versions of these actions and eventually once they supported v3, everything would be fine.

> However, it sounds like, based on some recent changes, this plan may not be as practical as we'd hoped, because I see that we've effectively restored Yarn v1 support for polyrepos to action-publish-release (https://github.com/MetaMask/action-publish-release/pull/83) and action-npm-publish (https://github.com/MetaMask/action-npm-publish/pull/51). (Unless I'm misunderstanding the intent of these PRs?)

For these PRs I assumed we want to keep supporting Yarn v1 for the time being, since some of our polyrepos use it. In MetaMask/action-publish-release#82 I introduced `yarn npm info`, but later realised this doesn't work for Yarn v1, so I changed it back in MetaMask/action-publish-release#83 to fix it. For the implementation in MetaMask/action-npm-publish#51 I simply copied the other implementation. The goal of those PRs was to support tags (since we release some packages in the Snaps repo under the `flask` tag), not to support Yarn v1 per se.

If we already broke Yarn v1 compatibility for these packages (I'm not sure if we did?) I suppose the PR to fix compatibility wasn't necessary. I didn't see anything that indicated Yarn v3 support only though. 

> However, if anyone wanted to extend this tool to support polyrepo packages, we would be back in the same boat we're in now, and we'd have to come up with a Yarn v1-compatible solution. In fact I believe we may already have one: [getWorkspaceLocations in action-utils](https://github.com/MetaMask/action-utils/blob/b33c0b6b5c6cb3d544f81231f6187a2925841ff5/src/package-utils.ts#L280). So, we could use this implementation instead of or even as an alternative to yarn workspaces list.

I hadn't thought about using this function. That seems like a better alternative!